### ISSUE TITLE
Improve Elasticsearch mappings

### ISSUE THREAD

**Issue by michaelwood** [ThreeSixtyGiving/grantnav#551]
The elasticsearch mappings in GrantNav could do with being looked at in more depth to make them more robust. Currently they are quite manually curated. This is related to #529 . 

Points for consideration:

- Map the whole 360Giving standard (possibly write/find a 'schema to mapping' translator?)
- Use more advanced features of elasticsearch such as dynamic templates to do the job more magically
- Handling custom fields from publishers (when to add to mapping, when to make searchable)
- Handling custom fields from additional data
- Creating a way to make sure what the datastore produces as GrantNav data doesn't diverge from what GrantNav is expecting.
- ...

**Comment by robredpath** [comment]
@michaelwood would there be a way to do this with the datastore? I feel like some combination of a 'schema to mapping' translator and some analysis of the additional fields that are present in any given data set fits better with the datastore than GN itself - given the current model. 

Mapping-as-a-Service, anyone?

**Comment by michaelwood** [comment]
> @michaelwood would there be a way to do this with the datastore? I feel like some combination of a 'schema to mapping' translator and some analysis of the additional fields that are present in any given data set fits better with the datastore than GN itself - given the current model.
> 
> Mapping-as-a-Service, anyone?

A 360 schema to elastic mapping generator would be good and logical! From the current state of the mappings though it suggests that there are quite a number of corner cases that have been added over time to support GrantNav's UI/UX so in reality it might be quite a bit more complicated. ...though I don't have the historical context to completely say that.

**Comment by robredpath** [comment]
>  there are quite a number of corner cases that have been added over time to support GrantNav's UI/UX 

Yeah, that's true; although nothing you couldn't work backwards from, I think. 

I've added a Trello card for the "automatically generate" bit as I think that's probably the most useful thing - both for GN, as we upgrade the schema, and potentially other people.
